sqlglot/dialects/dialect.py

```diff
@@ -571,6 +571,20 @@ def datestrtodate_sql(self: Generator, expression: exp.DateStrToDate) -> str:
     return self.sql(exp.cast(expression.this, "date"))
 
 
+# Used for Presto and Duckdb which use functions that don't support charset, and assume utf-8
+def encode_decode_sql(self: Generator, expression: exp.Expression, name: str) -> str:
+    if "charset" in expression.args:
+        charset = expression.args["charset"]
+
+        if charset.name.lower() != "utf-8":
+            self.unsupported(f"Unsupported charset {charset}")
+
+        expression = expression.copy()
+        del expression.args["charset"]
```

Comment:
I don't love this, happy to change this if there is a better way.

Reply:
I think you can just do `return self.func(name, expression.this, expression.args.get("replace") if replace else None)` (see other comment about `replace` flag).